iroh-util/src/lib.rs

```diff
@@ -0,0 +1,30 @@
+use std::{
+    cell::RefCell,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+
+/// Blocks current thread until ctrl-c is received
+pub async fn block_until_sigint() {
+    let (ctrlc_send, ctrlc_oneshot) = futures::channel::oneshot::channel();
+    let ctrlc_send_c = RefCell::new(Some(ctrlc_send));
```

Comment:
I'm curious why this is a `Refcell` and not an `Arc`. Or why it needs to be a cell type thing at all.